runtime/autoload/tar.vim

```diff
@@ -432,11 +433,14 @@ if v:shell_error != 0
   set nomod
 
   let &report= repkeep
+  exe "cd ".pwdkeep
```

Comment:
```suggestion
  cd -
```
and then get rid of `pwdkeep` var

Reply:
@chrisbra `cd -` might not do what is expected if `v:shell_error`:

```
if v:shell_error != 0
   cd ..
   call s:Rmdir("_ZIPVIM_")
   exe "cd ".fnameescape(curdir)
   echohl Error | echo "***error*** (tar#Read) sorry, unable to open or extract ".tarfile." with ".fname | echohl None
  endif
```

It would change it back to the original path and then `cd -` would change it back to the `tmpdir`.

So I would use `exe "lcd " .. fnameescape(curdir)` there.

Reply:
That is unfortunate we can't easily use `defer` with old lambdas:

```
" attempt to change to the indicated directory
try 
 exe "lcd ".fnameescape(tmpdir)
 defer {-> lcd -}
```

Reply:
yes makes sense. We could potentially also use a `try/catch/finally` block.